test/unit/test_disk_tensor.py

```diff
@@ -333,63 +333,79 @@ def test_copy_from_disk_huge(self):
       on_dev = t.to(Device.DEFAULT).realize()
       np.testing.assert_equal(on_dev.numpy(), t.numpy())
 
-class TestTarExtract(unittest.TestCase):
+class TestTarExtractPAX(unittest.TestCase):
+  tar_format = tarfile.PAX_FORMAT
+  max_link_len = 1000_000
+
   def setUp(self):
-    self.test_dir = tempfile.mkdtemp()
+    self.tar_path = pathlib.Path(tempfile.mktemp(".tar"))
     self.test_files = {
-      'file1.txt': b'Hello, World!',
-      'file2.bin': b'\x00\x01\x02\x03\x04',
-      'empty_file.txt': b''
+      'a/file1.txt': b'Hello, World!',
+      'a/b/file2.bin': b'\x00\x01\x02\x03\x04',
+      'empty_file.txt': b'',
+      'very'*15 + "/" + "very"*15 + '_long_filename.txt': b'Hello, World!'
```

Comment:
also test one that has more than 512 bytes

Reply:
I have added tests for filenames and data larger than 512 bytes.